pumps/pump.go

```diff
@@ -1,14 +1,17 @@
 package pumps
 
 import (
+	"context"
 	"errors"
 )
 
 type Pump interface {
 	GetName() string
 	New() Pump
 	Init(interface{}) error
-	WriteData([]interface{}) error
+	WriteData(context.Context, []interface{}) error
+	SetTimeout(timeout int)
+	GetTimeout() int
```

Comment:
remove this because the ctx passed to `WriteData` already comes with the deadline pumps should use it to know if the deadline is over (That is the proper use of `context.Context` )